client/web/src/tracking/eventLogger.ts

```diff
@@ -113,6 +114,21 @@ export class EventLogger implements TelemetryService {
         return firstSourceURL
     }
 
+    public getReferrer(): string {
+        const referrer = document.referrer
+        try {
+            // 🚨 SECURITY: Return if the referrer is a valid non-Sourcegraph.com URL
+            // to avoid leaking private code URLs.
+            const url = new URL(referrer)
+            if (url.hostname !== 'sourcegraph.com') {
```

Comment:
Isn't this the wrong way around? Currently it only returns referrers if the source is not sourcegraph cloud 

```suggestion
            if (url.hostname === 'sourcegraph.com') {
```

Reply:
That's correct! We want to know the URL a user comes from if they're coming from outside Sourcegraph Cloud. The use case is for product to know which sites (GitHub, Google, etc.) a user comes from before landing on Sourcegraph and whether that changes use behavior. The reason we don't want to track internal URLs is that if they switch pages from a private Sourcegraph Cloud repo to another page, we don't want to send that private URL back to our data stores.

Reply:
Discussed via slack, this is not to disable it on premise but to hide private repo names.